src/ripple/app/ledger/LedgerHistory.h

```diff
@@ -61,7 +61,7 @@ class LedgerHistory
 
     /** Retrieve a ledger given its hash */
     std::shared_ptr<Ledger const>
-    getLedgerByHash(LedgerHash const& ledgerHash);
+    getLedgerByHash(LedgerHash const& ledgerHash, bool acquire = true);
```

Comment:
The implications of `bool acquire` are not at all obvious to the casual reader.  You should probably add documentation of this parameter to the doxygen.